content-service/src/contentReplicator.js

```diff
@@ -23,6 +23,87 @@ class ContentReplicator {
   }
 
+  async _replicateTrack (track) {
+    const trackId = track.track_id
+    try {
+      const blocknumber = parseInt(track.blocknumber)
+      let start = Date.now()
+      let numMultihashes = 0
+      if (track.cover_art) {
+        await this._replicateTrackMultihash(track.track_id, track.cover_art, true)
+        numMultihashes++
+      }
+      if (track.metadata_multihash) {
+        await this._replicateTrackMultihash(track.track_id, track.metadata_multihash, true)
+        numMultihashes++
+      }
+
+      if (track.track_segments) {
+        let segments = track.track_segments
+        logger.info(`TrackID ${trackId} - ${segments.length} total segments`)
+        for (let i = 0; i < segments.length; i += 10) {
+          const slice = segments.slice(i, i + 10)
+          logger.info(`TrackID ${trackId} - Processing segments ${i} to ${i + 10}`)
+          await Promise.all(
+            slice.map(segment => {
+              return this._replicateTrackMultihash(trackId, segment.multihash)
+            })
+          )
+          numMultihashes += slice.length
+        }
+      }
+
+      let currentHighestTrackBlock = await this.queryHighestBlockNumber('track')
+      if (blocknumber > currentHighestTrackBlock) {
+        await models.Block.create({
+          blocknumber: blocknumber,
+          type: 'track'
+        })
+      }
+      let duration = Date.now() - start
+      logger.info(`TrackID ${trackId} - Replication complete. ${numMultihashes} multihashes in ${duration}ms`)
+    } catch (e) {
+      logger.error(`TrackID ${trackId} - ERROR PROCESSING ${e}`)
+      throw e
+    }
+  }
+
+  async _replicateUser (user) {
+    try {
+      let start = Date.now()
+      let numMultihashes = 0
+      const blocknumber = parseInt(user.blocknumber)
+      const userPinPromises = []
+      if (user.profile_picture) {
+        userPinPromises.push(this._replicateUserMultihash(user.user_id, user.profile_picture))
+        numMultihashes++
+      }
+      if (user.cover_photo) {
+        userPinPromises.push(this._replicateUserMultihash(user.user_id, user.cover_photo))
+        numMultihashes++
+      }
+      if (user.metadata_multihash) {
+        userPinPromises.push(this._replicateUserMultihash(user.user_id, user.metadata_multihash))
+        numMultihashes++
+      }
+
+      let currentHighestUserBlock = await this.queryHighestBlockNumber('user')
+      if (blocknumber > currentHighestUserBlock) {
+        await models.Block.create({
+          blocknumber: blocknumber,
+          type: 'user'
+        })
+      }
+      let duration = Date.now() - start
+      if (numMultihashes > 0) {
+        logger.info(`User ${user.user_id} - Replication complete. ${numMultihashes} multihashes in ${duration}ms`)
+      }
+    } catch (e) {
+      logger.info(`UserD ${user.user_id} - ERROR PROCESSING ${e}`)
+      throw e
+    }
+  }
+
   async _replicateTrackMultihash (pinTrackId, pinMultihash) {
```

Comment:
bit of a nit but looks like replicateTrackMultihash and replicateUserMultihash are the same code except that the File table query is done on trackId and userId. feels like could combine this function and pass in a type instead?

Reply:
agreed - can get this in prior to merging

Reply:
fixed in my PR